pkg/plugin/plugin.go

```diff
@@ -280,10 +319,30 @@ func CmdAdd(args *skel.CmdArgs) error {
 	}
 
 	var vlanTagNum uint = 0
+	var vlanMode = "access"
 	trunks := make([]uint, 0)
-	portType := "access"
-	if netconf.VlanTag == nil || len(netconf.Trunk) > 0 {
-		portType = "trunk"
+	cvlans := make([]uint, 0)
+
+	if netconf.VlanMode != nil {
+		vlanMode = *netconf.VlanMode
+	}
+
+	if netconf.VlanTag != nil && netconf.Cvlan != nil {
+		if vlanMode != "dot1q-tunnel" {
```

Comment:
what if vlanMode is not specified and cvlan is set in the NAD ? 
Let's set vlanMode to dot1q-tunnel when values are specified for cvlan.